tool.go

```diff
@@ -636,12 +636,14 @@ func (fs serveCommandFileSystem) Open(requestName string) (http.File, error) {
 		if err == nil {
 			return f, nil
 		}
+		defer f.Close()
```

Comment:
This is not valid for two reasons:

1. The `if` statement above checks if `err == nil`. If so, it returns `f, nil`.

    Otherwise, execution continues and gets to this point. We know that `err != nil` must be the case. That means `f` is not a valid file and there's no need to close it. In fact, it may be nil, so this could panic with nil pointer dereference here.

2. We are inside the `serveCommandFileSystem.Open(requestName string) (http.File, error)` method. It implements [`http.FileSystem`](https://godoc.org/net/http#FileSystem) interface. The `Open(name string) (File, error)` method, similar to [`os.Open`](https://godoc.org/os#Open), opens a file and returns it. It's the user's responsibility to close it when they are done using it. `Open` both cannot and should not try to close the file (otherwise, it's already closed by the time it's returned, which defeats the purpose).